ui/src/shared/components/ConfirmButton.js

```diff
@@ -0,0 +1,110 @@
+import React, {Component, PropTypes} from 'react'
+
+import OnClickOutside from 'shared/components/OnClickOutside'
+
+class ConfirmButton extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  handleButtonClick = () => {
+    if (this.props.disabled) {
+      return
+    }
+    this.setState({expanded: !this.state.expanded})
+  }
+
+  handleConfirmClick = () => {
+    this.setState({expanded: false})
+    this.props.confirmAction()
+  }
+
+  handleClickOutside = () => {
+    this.setState({expanded: false})
+  }
+
+  calculatePosition = () => {
```

Comment:
@alexpaxton is this for responsiveness?

Reply:
Yeah it basically prevents the confirmation tooltip from extending off the right edge of the screen. In the future I'll have to circle back and extend it to account for all screen edges to make it a truer component